packages/tldraw/src/lib/tools/SelectTool/selectHelpers.ts

```diff
@@ -34,38 +35,41 @@ export function kickoutOccludedShapes(editor: Editor, shapeIds: TLShapeId[]) {
 	}
 
 	// now kick out the children
-	// TODO: make this reparent to the parent's parent?
 	editor.reparentShapes(kickedOutChildren, editor.getCurrentPageId())
 }
 
 /** @internal */
 export function isShapeOccluded(editor: Editor, occluder: TLShape, shape: TLShapeId) {
 	const occluderPageBounds = editor.getShapePageBounds(occluder)
-	if (!occluderPageBounds) return false
-
 	const shapePageBounds = editor.getShapePageBounds(shape)
-	if (!shapePageBounds) return true
-
-	// If the shape's bounds are completely inside the occluder, it's not occluded
-	if (occluderPageBounds.contains(shapePageBounds)) {
-		return false
-	}
+	if (!occluderPageBounds) return true
```

Comment:
sorry, just to make sure i understand this part of the change:
now, the !occluderPageBounds returns true instead of false, and !shapePageBounds returns false instead of true. why? was that related to this fix, or unrelated?

Reply:
It's another think I tweaked in this PR!

It's a weird niche case: "What should happen when either shape has a size of zero?"
I think that, if a parent has zero size, then technically the child is *always* occluded, right?
And if a child has zero size, then it can't possibly be occluded, as it....... doesn't exist??? Weird one. I don't think this would every happen on tldraw.com

Reply:
haha, ok gotcha - i think that makes sense to me! agreed that it's sorta weird either way you slice it